servers/rendering/renderer_rd/storage_rd/render_scene_buffers_rd.h

```diff
@@ -0,0 +1,273 @@
+	RS::ViewportScreenSpaceAA screen_space_aa = RS::VIEWPORT_SCREEN_SPACE_AA_DISABLED;
+	bool use_taa = false;
+	bool use_debanding = false;
+
+	// Named Textures
+
+	struct NTKey {
+		StringName context;
+		StringName buffer_name;
+
+		bool operator==(const NTKey &p_val) const {
+			return (context == p_val.context) && (buffer_name == p_val.buffer_name);
+		}
+
+		static uint32_t hash(const NTKey &p_val) {
+			// FIXME, properly hash two stringnames together
+			uint32_t h = p_val.context.hash();
+			h = hash_murmur3_one_32(p_val.buffer_name.hash(), h);
+			return hash_fmix32(h);
+		}
+
+		NTKey() {}
+		NTKey(const StringName p_context, const StringName p_texture_name) {
+			context = p_context;
+			buffer_name = p_texture_name;
+		}
+	};
+
+	struct NamedTexture {
+		// Cache the data used to create our texture
+		RD::TextureFormat format;
+		bool is_unique; // If marked as unique, we return it into our pool
+
+		// Our texture objects, slices are lazy (i.e. only created when requested).
+		RID texture;
+		Vector<RID> slices;
+	};
+
+	mutable HashMap<NTKey, NamedTexture, NTKey> named_textures;
+	void free_named_texture(NamedTexture &p_named_texture);
+
+	// Data buffers
+	mutable HashMap<StringName, Ref<RenderSceneBufferDataRD>> data_buffers;
+
+protected:
+	static void _bind_methods();
+
+public:
+	RenderSceneBuffersRD();
+	virtual ~RenderSceneBuffersRD();
+
+	// info from our renderer
+	void set_can_be_storage(const bool p_can_be_storage) { can_be_storage = p_can_be_storage; }
+	void set_max_cluster_elements(const uint32_t p_max_elements) { max_cluster_elements = p_max_elements; }
+	void set_base_data_format(const RD::DataFormat p_base_data_format) { base_data_format = p_base_data_format; }
+	RD::DataFormat get_base_data_format() const { return base_data_format; }
+	void set_sseffects(RendererRD::SSEffects *p_ss_effects) { sse = p_ss_effects; }
+	void set_vrs(RendererRD::VRS *p_vrs) { vrs = p_vrs; }
+
+	void cleanup();
+	virtual void configure(RID p_render_target, const Size2i p_internal_size, const Size2i p_target_size, float p_fsr_sharpness, float p_texture_mipmap_bias, RS::ViewportMSAA p_msaa, RenderingServer::ViewportScreenSpaceAA p_screen_space_aa, bool p_use_taa, bool p_use_debanding, uint32_t p_view_count) override;
+	virtual void set_fsr_sharpness(float p_fsr_sharpness) override;
+	virtual void set_texture_mipmap_bias(float p_texture_mipmap_bias) override;
+	virtual void set_use_debanding(bool p_use_debanding) override;
+
+	// Named Textures
+
+	bool has_texture(const StringName &p_context, const StringName &p_texture_name) const;
+	RID create_texture(const StringName &p_context, const StringName &p_texture_name, const RD::DataFormat p_data_format, const uint32_t p_usage_bits, const RD::TextureSamples p_texture_samples = RD::TEXTURE_SAMPLES_1, const Size2i p_size = Size2i(0, 0), const uint32_t p_depth = 1, const uint32_t p_layers = 0, const uint32_t p_mipmaps = 1, RD::TextureView p_view = RD::TextureView(), bool p_unique = true);
```

Comment:
Feel free to pass `RD::TextureFormat`, we have RDTextureFormat object that can be used for the binder specific version, this can be added later on when exposed.

Reply:
The problem we have at the moment is that in many cases we have the RD::DataFormat, not the RD::TextureFormat. So we would need to do a little more work to make sure we use TextureFormat more consistantly.

Reply:
I have no idea when we use this, but in this case you definitely want to pass TextureFormat, as this may change in the future versions of Godot and force us to modify this API and break compatibility.

Reply:
This all became an issue when we introduced the mobile renderer and many of the buffers needed to be 32bit buffers instead of 64bit ones. And we're already using RD::DataFormat there.

But in hindsight, we probably should have used RD::TextureFormat there as well. I'll see how easy it is to change over fully.

Reply:
Ah hang on, the penny drops. You want to pass the `RD::TextureFormat` struct in its entirety? For some reason I had in my head that you wanted to change this to `Image::Format` or something similar that was already exposed to GDScript...

This would greatly bloat a lot of the code where we now have a single line solution but I can see the use case for having it. 

For now I've changed it so that we have a:
`RID create_texture_from_format(const StringName &p_context, const StringName &p_texture_name, const RD::TextureFormat &p_texture_format, RD::TextureView p_view = RD::TextureView(), bool p_unique = true);`
That takes a texture format struct as input.

But I've left:
`RID create_texture(const StringName &p_context, const StringName &p_texture_name, const RD::DataFormat p_data_format, const uint32_t p_usage_bits, const RD::TextureSamples p_texture_samples = RD::TEXTURE_SAMPLES_1, const Size2i p_size = Size2i(0, 0), const uint32_t p_layers = 0, const uint32_t p_mipmaps = 1, bool p_unique = true);`
in place and this is what is called in most places with most properties left to default.
Internally it builds the correct texture format and calls `create_texture_from_format`.

This gives us a best of both world solution, it keeps the API lean in 99% of the cases, and in the 1% where we need to define a specific texture format, we can do so.